src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -85,21 +86,29 @@ private Address buildAddress(OfficeAddress dwpAddress) {
     }
 
     public String getDwpRegionalCenterByBenefitTypeAndOffice(String benefitType, String dwpIssuingOffice) {
-
         Optional<OfficeMapping> officeMapping = getDwpMappingByOffice(benefitType, dwpIssuingOffice);
-
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
     }
 
     public String getDefaultDwpRegionalCenterByBenefitTypeAndOffice(String benefitType) {
-
         Optional<OfficeMapping> officeMapping = getDefaultDwpMappingByBenefitType(benefitType);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
+    }
 
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+    private String getDwpRegionalCenterByBenefitType(String benefitType, Optional<OfficeMapping> officeMapping) {
+        Benefit benefit;
+        try {
+            benefit = Benefit.getBenefitByCode(benefitType);
+        } catch (BenefitMappingException e) {
```

Comment:
It may be worth writing an api that returns an Option `Benefit.getBenefitByCodeOpt(benefitType)`